### PR TITLE
Add tests for output model files

### PR DESCRIPTION
**Pull request by rlh1994**
- Add in testing that runs the final file with a mock input and compares to the expected output (minus whitespace) to attempt to avoid failure due to changes in the core file that previously was not tested
- Fix bug with double empty string when sde is not provided
- Force sort on flat columns to ensure stability for testing
- Fix newlines and spelling in other files

### CODE REVIEW

File: utils/tests/test_split_event_config.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
+{
+    "config":{
+        "resolver_file_path": "default",
+        "filtered_events_table_name":"test_split_events",
```
Comment (emielver) [inline utils/tests/test_split_event_config.json:4]: Are we changing this filename with the rename?

File: utils/tests/test_model_gen.py [hunk 2]
```diff
@@ -534,4 +543,77 @@ def test_accept_del_all(self, setup_teardown, monkeypatch, capfd):
         expected_files = setup_teardown.get('file_names')[0:5]
         assert re.match(r'^Cleanup will remove models: {.*}\s*$', out.split('\n')[0])
         assert re.match(r'^Deleted 22 models, quitting...\s*$', out.split('\n')[1])
-        assert set(files) == set(expected_files)
\ No newline at end of file
+        assert set(files) == set(expected_files)
+
+class Test_model_output:
```
Comment (emielver) [inline utils/tests/test_model_gen.py:548]: Sorry if this is a dumb question but where is this being referenced?

**Review by emielver — APPROVED** [review]
LGTM! Couple of questions but nothing that is blocking!

File: utils/tests/test_split_event_config.json (hunk 1 above)
Comment (rlh1994) [reply utils/tests/test_split_event_config.json:4]: Yep!

File: utils/tests/test_model_gen.py (hunk 2 above)
Comment (rlh1994) [reply utils/tests/test_model_gen.py:548]: It all gets called in the GH Actions when pytest is run (or when you manually run pytest) - the class is to help cluster together similar tests (in this case all the tests of the actual output files), and also to use the fixture to run at the start of the tests, yield to running the tests part way through once the files exist, and then once the tests are completed it finishes running and removes all the files it created as we have to use the actual models folder so can't use a temp dir that will get dropped anwyay. 

The slight difference is this doesn't do anything to explicitely increase tests coverage because we're only looking at coverage of lines in the functions file - this tests the main script as black box based on the output, but doesn't refer to any specific part of that script to test individual parts.

That make sense?

Comment (emielver) [reply utils/tests/test_model_gen.py:548]: Yep, makes sense. Thanks for the additional info!
